### ISSUE TITLE
setOverrideHostAndPort not respected when sending OPTIONS response 

### ISSUE THREAD

**Issue by rjjkm** [resiprocate/resiprocate#168]
After calling setOverrideHostAndPort on the UAS, the default values (not overrides) are used in the Contact header when responding to an OPTIONS request. This is problematic when running resiprocate in a containerized enviroment where the local IP address is non routable externally and therefore should not be included in the SIP messages.  I realize OPTIONS is out of dialog however advertising inside IP addresses is not desirable.   

To test the following code was added to BasicCall.cxx:661  

```cpp
resip::Uri hostPort;
hostPort.host() = "10.10.10.8";
hostPort.port() = 12010;
dumUas->getMasterProfile()->setOverrideHostAndPort(hostPort);
```

UAS Contact header from test:
<sip:127.0.0.1:12010>

Is this behavior expected for OPTIONS? It works fine for other messages.

Thank you

**Comment by sgodin** [comment]
Sorry for the slow response.  We will leave this issue open, until someone has a chance to investigate.
Scott

**Comment by cyr2242** [comment]
Hi, IMHO the contact header in responses from the UAS carries the uri to be used in the following requests but only for the current dialog. Due the OPTIONS requests are, by default, out of a dialog, it won't have any sense to answer with a different uri if the first OPTIONS request was able to reach, unless you want to comunicate, for example, a redirection with a 3XX response: [https://datatracker.ietf.org/doc/html/rfc3261#section-11.2](https://datatracker.ietf.org/doc/html/rfc3261#section-11.2). In this case, the overwriting of the contact header in responses to OPTIONS might be allowed, therefore the comments: "_Contact header is not required for Requests that do not form a dialog_" at line 1091 of Dialog.cxx and "_in general, this should not create a Contact header since only requests ..._ "  at line 359 of Helper.cxx (_Helper::makeResponse_ method) are nearly a little wrong and maybe it would be possible to do some kind of treatment with the contact header in the response to OPTIONS at that point. Just my 2 cents. Regards.

**Comment by sgodin** [comment]
I agree it should use the override host and port from the master profile.
We would gladly accept a pull request fix for this, otherwise it will need
to wait for someone with some cycles.  :)

Thanks for posting the workaround.

Scott
